File: cpp/Solitaire/DeckTests/value_tests.cpp
#include "CppUnitTest.h"

#include <silnith/game/deck/value.h>

using namespace Microsoft::VisualStudio::CppUnitTestFramework;

using namespace silnith::game::deck;

using namespace std::literals::string_literals;

template<>
std::wstring Microsoft::VisualStudio::CppUnitTestFramework::ToString<>(value const& value)
{
	return to_wstring(value);
}

namespace DeckTests
{
	TEST_CLASS(ValueTests)
	{
	public:
		TEST_METHOD(TestGetValueAce)
		{
			Assert::AreEqual(1, get_value(value::ace));
		}

		TEST_METHOD(TestGetValueTwo)
		{
			Assert::AreEqual(2, get_value(value::two));
		}

		TEST_METHOD(TestGetValueThree)
		{
			Assert::AreEqual(3, get_value(value::three));
		}

		TEST_METHOD(TestGetValueFour)
		{
			Assert::AreEqual(4, get_value(value::four));
		}

		TEST_METHOD(TestGetValueFive)
		{
			Assert::AreEqual(5, get_value(value::five));
		}

		TEST_METHOD(TestGetValueSix)
		{
			Assert::AreEqual(6, get_value(value::six));
		}

		TEST_METHOD(TestGetValueSeven)
		{
			Assert::AreEqual(7, get_value(value::seven));
		}

		TEST_METHOD(TestGetValueEight)
		{
			Assert::AreEqual(8, get_value(value::eight));
		}

		TEST_METHOD(TestGetValueNine)
		{
			Assert::AreEqual(9, get_value(value::nine));
		}

		TEST_METHOD(TestGetValueTen)
		{
			Assert::AreEqual(10, get_value(value::ten));
		}

		TEST_METHOD(TestGetValueJack)
		{
			Assert::AreEqual(11, get_value(value::jack));
		}

		TEST_METHOD(TestGetValueQueen)
		{
			Assert::AreEqual(12, get_value(value::queen));
		}

		TEST_METHOD(TestGetValueKing)
		{
			Assert::AreEqual(13, get_value(value::king));
		}

		TEST_METHOD(TestToStringAce)
		{
			Assert::AreEqual("ace"s, to_string(value::ace));
		}

		TEST_METHOD(TestToStringTwo)
		{
			Assert::AreEqual("two"s, to_string(value::two));
		}

		TEST_METHOD(TestToStringThree)
		{
			Assert::AreEqual("three"s, to_string(value::three));
		}

		TEST_METHOD(TestToStringFour)
		{
			Assert::AreEqual("four"s, to_string(value::four));
		}

		TEST_METHOD(TestToStringFive)
		{
			Assert::AreEqual("five"s, to_string(value::five));
		}

		TEST_METHOD(TestToStringSix)
		{
			Assert::AreEqual("six"s, to_string(value::six));
		}

		TEST_METHOD(TestToStringSeven)
		{
			Assert::AreEqual("seven"s, to_string(value::seven));
		}

		TEST_METHOD(TestToStringEight)
		{
			Assert::AreEqual("eight"s, to_string(value::eight));
		}

		TEST_METHOD(TestToStringNine)
		{
			Assert::AreEqual("nine"s, to_string(value::nine));
		}

		TEST_METHOD(TestToStringTen)
		{
			Assert::AreEqual("ten"s, to_string(value::ten));
		}

		TEST_METHOD(TestToStringJack)
		{
			Assert::AreEqual("jack"s, to_string(value::jack));
		}

		TEST_METHOD(TestToStringQueen)
		{
			Assert::AreEqual("queen"s, to_string(value::queen));
		}

		TEST_METHOD(TestToStringKing)
		{
			Assert::AreEqual("king"s, to_string(value::king));
		}

		TEST_METHOD(TestToStringInvalidValue)
		{
			Assert::AreEqual("14"s, to_string(static_cast<value>(14)));
		}

		TEST_METHOD(TestToWStringAce)
		{
			Assert::AreEqual(L"ace"s, to_wstring(value::ace));
		}

		TEST_METHOD(TestToWStringTwo)
		{
			Assert::AreEqual(L"two"s, to_wstring(value::two));
		}

		TEST_METHOD(TestToWStringThree)
		{
			Assert::AreEqual(L"three"s, to_wstring(value::three));
		}

		TEST_METHOD(TestToWStringFour)
		{
			Assert::AreEqual(L"four"s, to_wstring(value::four));
		}

		TEST_METHOD(TestToWStringFive)
		{
			Assert::AreEqual(L"five"s, to_wstring(value::five));
		}

		TEST_METHOD(TestToWStringSix)
		{
			Assert::AreEqual(L"six"s, to_wstring(value::six));
		}

		TEST_METHOD(TestToWStringSeven)
		{
			Assert::AreEqual(L"seven"s, to_wstring(value::seven));
		}

		TEST_METHOD(TestToWStringEight)
		{
			Assert::AreEqual(L"eight"s, to_wstring(value::eight));
		}

		TEST_METHOD(TestToWStringNine)
		{
			Assert::AreEqual(L"nine"s, to_wstring(value::nine));
		}

		TEST_METHOD(TestToWStringTen)
		{
			Assert::AreEqual(L"ten"s, to_wstring(value::ten));
		}

		TEST_METHOD(TestToWStringJack)
		{
			Assert::AreEqual(L"jack"s, to_wstring(value::jack));
		}

		TEST_METHOD(TestToWStringQueen)
		{
			Assert::AreEqual(L"queen"s, to_wstring(value::queen));
		}

		TEST_METHOD(TestToWStringKing)
		{
			Assert::AreEqual(L"king"s, to_wstring(value::king));
		}

		TEST_METHOD(TestToWStringInvalidValue)
		{
			Assert::AreEqual(L"14"s, to_wstring(static_cast<value>(14)));
		}
	};
}

File: csharp/Solitaire/DeckTests/ValueTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Diagnostics.CodeAnalysis;

namespace Silnith.Game.Deck.Tests
{
    [TestClass]
    [SuppressMessage("Design", "MSTEST0032:Assertion condition is always true", Justification = "The unit test is making sure the enum is defined properly.")]
    public class ValueTests
    {
        [TestMethod]
        public void TestGetValueAce()
        {
            Assert.AreEqual(1, Value.Ace.GetValue());
        }

        [TestMethod]
        public void TestGetValueTwo()
        {
            Assert.AreEqual(2, Value.Two.GetValue());
        }

        [TestMethod]
        public void TestGetValueThree()
        {
            Assert.AreEqual(3, Value.Three.GetValue());
        }

        [TestMethod]
        public void TestGetValueFour()
        {
            Assert.AreEqual(4, Value.Four.GetValue());
        }

        [TestMethod]
        public void TestGetValueFive()
        {
            Assert.AreEqual(5, Value.Five.GetValue());
        }

        [TestMethod]
        public void TestGetValueSix()
        {
            Assert.AreEqual(6, Value.Six.GetValue());
        }

        [TestMethod]
        public void TestGetValueSeven()
        {
            Assert.AreEqual(7, Value.Seven.GetValue());
        }

        [TestMethod]
        public void TestGetValueEight()
        {
            Assert.AreEqual(8, Value.Eight.GetValue());
        }

        [TestMethod]
        public void TestGetValueNine()
        {
            Assert.AreEqual(9, Value.Nine.GetValue());
        }

        [TestMethod]
        public void TestGetValueTen()
        {
            Assert.AreEqual(10, Value.Ten.GetValue());
        }

        [TestMethod]
        public void TestGetValueJack()
        {
            Assert.AreEqual(11, Value.Jack.GetValue());
        }

        [TestMethod]
        public void TestGetValueQueen()
        {
            Assert.AreEqual(12, Value.Queen.GetValue());
        }

        [TestMethod]
        public void TestGetValueKing()
        {
            Assert.AreEqual(13, Value.King.GetValue());
        }

        [TestMethod]
        public void TestAceLessThanTwo()
        {
            Assert.IsTrue(Value.Ace < Value.Two);
        }

        [TestMethod]
        public void TestTwoGreaterThanAce()
        {
            Assert.IsTrue(Value.Two > Value.Ace);
        }

        [TestMethod]
        public void TestTenLessThanJack()
        {
            Assert.IsTrue(Value.Ten < Value.Jack);
        }

        [TestMethod]
        public void TestJackGreaterThanTen()
        {
            Assert.IsTrue(Value.Jack > Value.Ten);
        }

        [TestMethod]
        public void TestJackLessThanQueen()
        {
            Assert.IsTrue(Value.Jack < Value.Queen);
        }

        [TestMethod]
        public void TestQueenGreaterThanJack()
        {
            Assert.IsTrue(Value.Queen > Value.Jack);
        }

        [TestMethod]
        public void TestQueenLessThanKing()
        {
            Assert.IsTrue(Value.Queen < Value.King);
        }

        [TestMethod]
        public void TestKingGreaterThanQueen()
        {
            Assert.IsTrue(Value.King > Value.Queen);
        }

        [TestMethod]
        public void TestToStringAce()
        {
            Assert.AreEqual("Ace", Value.Ace.ToString());
        }

        [TestMethod]
        public void TestToStringTwo()
        {
            Assert.AreEqual("Two", Value.Two.ToString());
        }

        [TestMethod]
        public void TestToStringThree()
        {
            Assert.AreEqual("Three", Value.Three.ToString());
        }

        [TestMethod]
        public void TestToStringFour()
        {
            Assert.AreEqual("Four", Value.Four.ToString());
        }

        [TestMethod]
        public void TestToStringFive()
        {
            Assert.AreEqual("Five", Value.Five.ToString());
        }

        [TestMethod]
        public void TestToStringSix()
        {
            Assert.AreEqual("Six", Value.Six.ToString());
        }

        [TestMethod]
        public void TestToStringSeven()
        {
            Assert.AreEqual("Seven", Value.Seven.ToString());
        }

        [TestMethod]
        public void TestToStringEight()
        {
            Assert.AreEqual("Eight", Value.Eight.ToString());
        }

        [TestMethod]
        public void TestToStringNine()
        {
            Assert.AreEqual("Nine", Value.Nine.ToString());
        }

        [TestMethod]
        public void TestToStringTen()
        {
            Assert.AreEqual("Ten", Value.Ten.ToString());
        }

        [TestMethod]
        public void TestToStringJack()
        {
            Assert.AreEqual("Jack", Value.Jack.ToString());
        }

        [TestMethod]
        public void TestToStringQueen()
        {
            Assert.AreEqual("Queen", Value.Queen.ToString());
        }

        [TestMethod]
        public void TestToStringKing()
        {
            Assert.AreEqual("King", Value.King.ToString());
        }
    }
}
